assets/questions/palindrome.js

```diff
@@ -0,0 +1,152 @@
+ */
+
+globalData.questions['palindrome'] = {  // eslint-disable-line dot-notation
+  title: 'Balanced Palindrome',
+  starterCode: {
+    python:
+`def isPalindrome(s):
+    return False
+`
+  },
+  auxiliaryCode: {
+    python:
+`class AuxiliaryCode(object):
+    @classmethod
+    def createPalindromeString(cls, chr_set, input_size):
+        import random
+        ret = ""
+        while input_size > 0:
+            chr = random.choice(chr_set)
+            if input_size == 1:
+                ret += chr
+                input_size -= 1
+            else:
+                ret = chr + ret + chr
+                input_size -= 2
+        return ret
+
+    @classmethod
+    def countNumberOfChar(cls, s):
+        ascii_codes = {i:0 for i in range(255)}
+        for chr in s:
+            ascii_codes[ord(chr)] += 1
+        for code in ascii_codes:
+            if ascii_codes[code] % 2 is not 0:
+                return False
+        return True
+
+    @classmethod
+    def forgetIgnoreSpace(cls, s):
+        for i in range(len(s)/2):
+            if s[i] is not s[-i-1]:
+                return False
+        return True
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'For this question, you will implement the isPalindrom function. ',
+        'It takes a sentense as input and return True if the input is a ',
+        'palindrome and False if it is not.'
+      ].join(''),
+      [
+        'A palindrom is a word, phrase, or sequence that reads the same ',
+        'backward as forward, e.g., \'madam\' or \'nurses run\'.'
+      ].join(''),
+      'Note, the input string would only consist of alphabets from a-z and space.'
+    ],
+    prerequisiteSkills: ['Strings', 'String Manipulation'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'isPalindrome',
+    correctnessTests: [{
+      input: '',
+      allowedOutputs: [true]
+    }, {
+      input: 'nurses run',
+      allowedOutputs: [true]
+    }, {
+      input: 'a b   cba',
+      allowedOutputs: [true]
+    }, {
+      input: 'abcdcba',
+      allowedOutputs: [true]
+    }, {
+      input: 'abccba',
+      allowedOutputs: [true]
+    }, {
+      input: 'abab',
+      allowedOutputs: [false]
+    }, {
+      input: 'stressed desserts',
+      allowedOutputs: [true]
+    }, {
+      input: 'dioramas is samaroid',
+      allowedOutputs: [false]
+    }, {
+      input: 'dioramas is si samaroid',
+      allowedOutputs: [true]
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.countNumberOfChar',
+      messages: [
+        "Try running your code on 'abab' on paper. Did you expet that result?",
+        [
+          'Are you sure the sentence is actually palindrom? \'abba\' is palindrom, ',
+          'but \'abab\' is not.'
+        ].join(''),
+        [
+          "It looks like you're counting the number of character of the input string, ",
```

Comment:
number of characters in the input string and returning true if there are an even number of each. With strings like 'abab', however, this approach doesn't work. Try to update your code to avoid just counting the characters.